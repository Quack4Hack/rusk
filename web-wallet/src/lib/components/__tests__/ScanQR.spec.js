import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { ScanQR } from "..";

describe("ScanQR", () => {
  const baseProps = {
    scanner: undefined,
  };

  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("renders the ScanQR component", () => {
    const { container } = render(ScanQR, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });
});
