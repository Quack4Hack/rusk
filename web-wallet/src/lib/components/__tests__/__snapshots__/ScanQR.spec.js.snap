// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScanQR > renders the ScanQR component 1`] = `
<div
  class="scan s--VhvQ6ibSubn"
>
  <video
    class="s--VhvQ6ibSubn"
  >
    <track
      class="s--VhvQ6ibSubn"
      kind="captions"
    />
  </video>
   
  <div
    class="scan__highlight s--VhvQ6ibSubn"
  />
   
  <button
    class="dusk-button dusk-button--type--button dusk-button--variant--secondary dusk-button--size--small"
    type="button"
  >
    <span
      class="dusk-button__text"
    >
      CLOSE
    </span>
  </button>
  <!--&lt;Button&gt;-->
</div>
`;
